File: java/leetcode371.java
class Solution {
    public int getSum(int a, int b) {
        int SUM=0;
        int CARRY;
        while(b!=0){
            CARRY=(a&b)<<1;
            a=a^b;
            
            b=CARRY;
        }

        return a;
        
    }
}

File: python/leetcode371.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        mask=0xffffffff

        while b!=0:
            CARRY = (a&b)<<1
            a=(a^b) & mask
            b=CARRY & mask


        if a > mask//2:
            return ~(a^mask)
        else:
            return a